### ISSUE TITLE
Picker not working on macOS

### ISSUE THREAD

**Issue by andrewstellman** [dotnet/maui#21691]
### Description

The Picker control seems to be broken on macOS. 
On macOS, clicking on the Picker displays a dialog that shows the title and a "Done" button but doesn't display the items:

<img width="305" alt="image" src="https://github.com/dotnet/maui/assets/7516297/f61b0f8d-fc6f-4c74-a0bd-38e18aaf4d86">

Here's a video that shows it in action:

https://github.com/dotnet/maui/assets/7516297/04f022b5-1aa5-4ca1-9c7b-f2d569f72cc6

I'm running it on the latest 8.0 GA SDK:

```
andrewstellman@Andrews-MBP reproduce-maui-picker-issue % dotnet workload list

Installed Workload Id      Manifest Version      Installation Source
--------------------------------------------------------------------
maui                       8.0.7/8.0.100         SDK 8.0.200        

Use `dotnet workload search` to find additional workloads to install.

andrewstellman@Andrews-MBP reproduce-maui-picker-issue % dotnet build -t:Run -f net8.0-maccatalyst

Welcome to .NET 8.0!
---------------------
SDK Version: 8.0.203
```

This same code works fine on Windows and Android:

Windows:
<img width="1352" alt="image" src="https://github.com/dotnet/maui/assets/7516297/75149e96-9407-4adf-b528-82672b34e43e">

Android:
<img width="500" alt="image" src="https://github.com/dotnet/maui/assets/7516297/c40586ff-bcf1-47e0-af20-65ee1c2f92c8">

### Steps to Reproduce

Here's some simple code to reproduce the issue:

XAML
```xaml
            <Picker x:Name="BirdPicker" Title="Pick a bird" />

            <Button x:Name="AddBird" Text="Add a bird"
                Margin="0,0,0,20" SemanticProperties.Hint="Adds a bird"
                Clicked="AddBird_Clicked"/>

            <Label x:Name="Birds"
                    Padding="10" MinimumWidthRequest="150"
                    TextColor="White" BackgroundColor="DarkBlue" 
                    SemanticProperties.Description="Shows the added birds" />
```

C#
```C#
	public MainPage()
	{
		InitializeComponent();

		BirdPicker.ItemsSource = new List<string> {
			"Duck",
			"Pigeon",
			"Penguin",
			"Ostrich",
			"Owl"
		};
	}

	private void AddBird_Clicked(object sender, EventArgs e)
	{
		if (!String.IsNullOrEmpty(Birds.Text))
		{
			Birds.Text = Birds.Text + Environment.NewLine;
		}
		Birds.Text += BirdPicker.SelectedItem;
	}
```

### Link to public reproduction project repository

https://github.com/andrewstellman/reproduce-maui-picker-issue/

### Version with bug

8.0.3 GA

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

Unknown/Other

### Affected platforms

macOS

**Comment by drasticactions** [comment]
Duplicate of https://github.com/dotnet/maui/issues/18015

In short, the way the picker is constructed on Catalyst makes it very fragile. It consists of an inline text input field that spawns a UIAlert with a picker control inside it, and that UIPicker has no height. If you remove the title, that "fixes" the constraints on the hacked-in UIPicker and lets it show its items.

<img width="1012" alt="スクリーンショット 2024-04-08 14 49 20" src="https://github.com/dotnet/maui/assets/898335/ff5be621-8254-4c3c-a88f-78dc399bc932">

It doesn't affect other platforms because they use more traditional pickers. The underlying implementation for Catalyst should need to be fixed.

**Comment by ninachen03** [comment]
Verified this issue with Visual Studio for mac 17.6.10(build428) (MauiVersion 8.0.14 & 8.0.0.-rc.2.9530). Can not repro this issue on maccatalyst.
![image](https://github.com/dotnet/maui/assets/128687574/5fcad2e7-2370-4bf3-9888-c8c8fe903cc1)

**Comment by drasticactions** [comment]
@ninachen03 The issue is that the picker items do not appear. You're only seeing the title of the alert, but not the items that should be shown within it. Removing the title from the picker will show the items. The bug is still there.

**Comment by andrewstellman** [comment]
@drasticactions thanks so much for out the workaround. This was going to be a really annoying issue for the [5th edition of Head First C#](https://github.com/head-first-csharp/fifth-edition), which I'm just finishing up. At least now I can find a way to give people a way to get their code working!

We're in production, so hopefully I can fix the code before we go to press so I don't have to have them print a warning that says "Mac readers - there's a bug that keeps the Picker from working if the Title property is set." Seems a little unfortunate that the issue is closed as not planned. 😢

**Comment by drasticactions** [comment]
@andrewstellman It was closed as a duplicate of the issue I linked to, which is the same as yours, and is still open. That issue has this listed as a "regression" but it's complicated: the code itself for the picker didn't change. I believe it's macOS specific where a version bump "broke" this implementation (so if you run this on earlier versions of macOS, it should "work" correctly).

I say this in quotes because, IMO (I'm not on the MAUI team, I'm in .NET and work on other tooling within VS to help .NET, such as Hot Reload. I triage issues and give help where I can), because I think this implementation is very hacky (Implementing a UIAlert that to then jam in a UIPicker) that "works" but is not ideal to do. So it can be "fixed," but I honestly think it needs to be redone for Mac Catalyst. Especially because if you enable the Desktop Idiom for Mac Catalyst, this picker will blow up your app, because UIPicker doesn't exist in Mac Catalyst and will actively throw an exception saying not to use it.
